### PR TITLE
docs: demo dataset ingestion

### LINKED ISSUE #4248: [onboarding] ingest one-time of demo dataset
Hook into the one-time background process and ingest the demo dataset

### PR DESCRIPTION
**Pull request by cjunkin**
Using the `--scaffold-dataset` parameter results in the ingestion of dataset defined in the project/trace fixture that was passed in with `with-projects` or `with-trace-fixtures`.

resolves #4248, #4247, #4357

### CODE REVIEW

**Comment by cjunkin** [comment]
image

**Comment by cjunkin** [comment]
Should there be metadata? It seems like downloading as jsonl doesn't save the metadata from the original dataset

image

**Comment by cjunkin** [comment]
image
image

File: src/phoenix/server/app.py [hunk 1]
```diff
@@ -271,9 +282,9 @@ async def _run(self) -> None:
         Determines whether to load fixtures and handles them.
         """
         if await self._should_load_fixtures():
-            logger.info("Loading trace fixtures.")
+            logger.info("Loading demo trace fixtures...")
             await self._handle_tracing_fixtures()
-            logger.info("Finished loading fixtures.")
+            logger.info("Finished demo loading fixtures.")
```
Comment (mikeldking) [inline src/phoenix/server/app.py:287]: I don't think you intended this change, or at least the english is confusing

File: src/phoenix/server/app.py [hunk 2]
```diff
@@ -258,6 +268,7 @@ def __init__(
             get_trace_fixture_by_name(name) for name in tracing_fixture_names
         )
         self._force_fixture_ingestion = force_fixture_ingestion
+        self._scaffold_dataset = scaffold_dataset
```
Comment (mikeldking) [inline src/phoenix/server/app.py:271]: I'm a bit confused why this is singular. Can't you have multiple?

File: src/phoenix/server/main.py [hunk 3]
```diff
@@ -182,34 +184,50 @@ def _get_pid_file() -> Path:
             "database is new."
         ),
     )
+    serve_parser.add_argument(
+        "--scaffold-dataset",
```
Comment (mikeldking) [inline src/phoenix/server/main.py:188]: can we make this plural and support multiple?

File: src/phoenix/trace/utils.py [hunk 4]
```diff
@@ -18,6 +18,16 @@ def is_jsonl_file(file_path: str) -> bool:
...
 
 
+def is_csv_file(file_path: str) -> bool:
+    """
+    Check if the given file is a csv file.
+    """
+    file_extension = os.path.splitext(file_path)[-1]
+    if file_extension == ".csv":
+        return True
+    return False
```
Comment (mikeldking) [inline src/phoenix/trace/utils.py:28]: this utility feels out of place, though it looks like is_jsonl was added above. Feels largely WET in a way that's not quite composable in a nice way (e.g. it seems like you just need a function that parses a file extension...)

**Review by mikeldking — COMMENTED** [review]
Left some minor comments. Passing to @RogerHYang for final review

File: src/phoenix/server/main.py (hunk 3 above)
Comment (cjunkin) [reply src/phoenix/server/main.py:188]: Yes, this should be plural and already supports multiple datasets. 

I did want to ask though if we wanted to be able to scaffold the datasets independently of fixtures. Currently, the flag doesn't take in any inputs and instead enables the ingestion of datasets defined under trace fixtures that are inputted through `--with-projects` or `with-trace-fixtures`.

File: src/phoenix/trace/utils.py (hunk 4 above)
Comment (cjunkin) [reply src/phoenix/trace/utils.py:28]: Implemented a function `parse_file_extension` instead

File: src/phoenix/server/main.py (hunk 3 above)
Comment (mikeldking) [reply src/phoenix/server/main.py:188]: I see, yeah forgot these were a bit coupled. Let's just get this working and can re-visit. If so I don't think we even need a flag.

File: src/phoenix/trace/fixtures.py [hunk 5]
```diff
@@ -225,7 +254,7 @@ class TracesFixture:
 NAME_TO_TRACES_FIXTURE: Dict[str, TracesFixture] = {
     fixture.name: fixture for fixture in TRACES_FIXTURES
 }
-PROJ_NAME_TO_TRACES_FIXTURE = defaultdict(list)
+PROJ_NAME_TO_TRACES_FIXTURE: Dict[str, List[TracesFixture]] = defaultdict(list)
```
Comment (RogerHYang) [inline src/phoenix/trace/fixtures.py:257]: fyi. defaultdict has a type
```suggestion
PROJ_NAME_TO_TRACES_FIXTURE: DefaultDict[str, List[TracesFixture]] = defaultdict(list)
```

**Review by RogerHYang — APPROVED** [review]
lgtm

**Pull request merged**